src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -118,6 +121,24 @@ internal void Freeze()
 
         private bool IsFrozen => _frozen != 0;
 
+        /// <summary>
+        /// This is the language-specific type symbol representing the private
+        /// implementation details class. Setter uses Interlocked.CompareExchange so that the
+        /// language proxy is set atomically, therefore the client should always
+        /// use the value from the getter after setting.
+        /// </summary>
```

Comment:
Since the setter might not set the value, I think property with a setter is a bad API for this purpose. Instead of having a setter like that, please add a method SetLanguageProxy that would return the value as well, thus eliminating the need to read the property again.